src/install/validate_npm_package_name.zig

```diff
@@ -0,0 +1,238 @@
+//! This module provides the functionality offered by
+//! https://github.com/npm/validate-npm-package-name.
+//!
+//! It deviates from the original implementation in its string error messages. The difference is
+//! that the original implementation is able to preserve the original name of the package in its
+//! error messages while ours is not. This is a conscious decision to avoid allocations and it is
+//! documented in the JS tests.
+extern fn ModuleLoader__isBuiltin(data: [*]const u8, len: usize) bool;
+
+/// The maximum length of a package name, according the upstream npm implementation.
+pub const max_package_name_length = 214;
+
+pub const Error = enum {
+    EmptyName,
+    NameStartsWithDot,
+    NameStartsWithUnderscore,
+    NameSurroundedWithSpaces,
+    IllegalName,
+    NameContainsNonUrlSafeCharacters,
```

Comment:
enum are usually snake case in zig

Reply:
Even errors?

Reply:
“Real” errors as in `error.FooBar` are PascalCase, but this is just an enum called `Error`